arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -143,6 +143,44 @@ public inline fun <Error, A> iorNel(noinline combineError: (NonEmptyList<Error>,
   return ior(combineError, block)
 }
 
+/**
+ * Runs a computation [block] using [Raise], and return its outcome as [Ior].
+ * - [Ior.Right] represents success,
+ * - [Ior.Left] represents logical failure which made it impossible to continue,
+ * - [Ior.Both] represents that some logical failures were raised,
+ *   but it was possible to continue until producing a final value.
+ *
+ * This function re-throws any exceptions thrown within the [Raise] block.
+ *
+ * In both [Ior.Left] and [Ior.Both] cases, if more than one logical failure
+ * has been raised, they are combined using [combineError].
+ *
+ * Read more about running a [Raise] computation in the
+ * [Arrow docs](https://arrow-kt.io/learn/typed-errors/working-with-typed-errors/#running-and-inspecting-results).
+ */
+@ExperimentalRaiseAccumulateApi
+public inline fun <Error, A> iorAccumulate(noinline combineError: (Error, Error) -> Error, block: RaiseAccumulate<Error>.() -> A): Ior<Error, A> {
+  contract { callsInPlace(block, InvocationKind.AT_MOST_ONCE) }
+  val state: Atomic<Any?> = Atomic(EmptyValue)
+  return merge {
+    val raiseAccumulated = RaiseAccumulate.Error { raise(Ior.Left(EmptyValue.unbox(state.get()))) }
+    val accumulate = object : Accumulate<Error> {
```

Comment:
I think this should go as part of `IorRaise`, which maybe should get an `IorRaiseAccumulate` subclass.